core-data/resolvers.js

```diff
@@ -17,3 +17,14 @@ export async function* getCategories() {
 	const categories = await apiRequest( { path: '/wp/v2/categories' } );
 	yield receiveTerms( 'categories', categories );
 }
+
+/**
+ * Requests a media element from the REST API.
+ *
+ * @param {Object} state State tree
+ * @param {number} id    Media id
+ */
+export async function* getMedia( state, id ) {
+	const media = await apiRequest( { path: `/wp/v2/media/${ id }` } );
+	yield receiveMedia( media );
```

Comment:
I'm not tracking any "is requesting" state at the moment because it's not really needed for our current use-cases. But I was wondering if we should have a separate `requests` state tree to keep track in a generic way of these states regardless of the request data type, query, ... 

Same here, I guess we'll see with use-cases while adding other resolvers.

Reply:
>But I was wondering if we should have a separate requests state tree to keep track in a generic way of these states regardless of the request data type, query, ...

The original implementation in ffb076057c577110ddf5d7140fe54eb53dc29300 had a standalone `requested` state, which was meant to be generic and may still be an option to explore. The idea of not having a dedicated state works particularly well when we adopt a pattern of a `null` value meaning that it's known to _exist_, but not yet having received the entities. Though this works only with collections. With singular entities, a `null` value may instead mean having received an empty entity (404?). I agree we'll have to see how it plays out.

Related: https://github.com/WordPress/gutenberg/pull/5219#issuecomment-373770328